src/common.c

```diff
@@ -151,24 +151,34 @@ size_t hexlenl(const char *q, int *extra_chars)
 	return _hexlen(q, atoi16l, extra_chars);
 }
 
+static int isdec_len(const char *q, const char *mxv)
+{
+	const char *p = q;
+	do
+	{
+		if (*p < '0' || *p > '9' || p - q >= 10)
+			return 0;
+	} while (*++p);
+
+	return p - q < 10 || strcmp(q, mxv) <= 0;
+}
 int isdec(const char *q)
 {
-	char buf[24];
-	int x = atoi(q);
-	sprintf(buf, "%d", x);
-	return !strcmp(q,buf) && *q != '-';
+	const char *imax = "2147483647";
+	return isdec_len(q, imax);
```

Comment:
No need to define `imax`, you can use simply:

    return isdec_len(q, "2147483647");